### PR TITLE
[CI] Update the pinned timm version

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #108076

Summary: Unify the pinned timm version and install timm at the docker building time

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

**Comment by ezyang** [comment]
nyeh, how come version instead of hash? Hash is a bit more flexible

**Comment by desertfire** [comment]
> nyeh, how come version instead of hash? Hash is a bit more flexible

Thought version number is easier for OSS users to refer to and compare with, but I can change it back.

File: .ci/docker/common/install_inductor_benchmark_deps.sh [hunk 1]
```diff
@@ -15,10 +15,12 @@ function install_timm() {
   local commit
   commit=$(get_pinned_commit timm)
   pip_install pandas==2.0.3
-  pip_install "git+https://github.com/rwightman/pytorch-image-models@${commit}"
+  pip_install "git+https://github.com/huggingface/pytorch-image-models@${commit}"
+  # Clean up
+  conda_run pip uninstall -y cmake torch torchvision triton
```
Comment (ezyang) [inline .ci/docker/common/install_inductor_benchmark_deps.sh:20]: 🤔

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**